<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE html SYSTEM "about:legacy-compat">
<html xmlns="http://www.w3.org/1999/xhtml" lang="en">
  <head>
    <title>phpDox - Symfony\Component\OptionsResolver\Exception\NoSuchOptionException</title>
    <link rel="stylesheet" type="text/css" href="../css/style.css" media="screen"/>
    <meta http-equiv="content-type" content="text/html; charset=utf-8"/>
  </head>
  <body>
    <nav class="topnav">
      <ul>
        <li>
          <div class="logo"><span>/**</span>phpDox</div>
        </li>
        <li class="separator">
          <a href="../index.xhtml">Overview</a>
        </li>
        <li class="separator">
          <a href="../namespaces.xhtml">Namespaces</a>
        </li>
        <li>
          <a href="../interfaces.xhtml">Interfaces</a>
        </li>
        <li>
          <a href="../classes.xhtml">Classes</a>
        </li>
        <li>
          <a href="../traits.xhtml">Traits</a>
        </li>
        <li class="separator">
          <a href="../source/index.xhtml">Source</a>
        </li>
      </ul>
    </nav>
    <div id="mainstage">
      <div class="box">
        <ul class="breadcrumb">
          <li>
            <a href="../index.xhtml">Overview</a>
          </li>
          <li class="separator">
            <a href="../classes.xhtml">Classes</a>
          </li>
          <li class="separator">
            <a href="../classes.xhtml#Symfony_Component_OptionsResolver_Exception">Symfony\Component\OptionsResolver\Exception</a>
          </li>
          <li class="separator">NoSuchOptionException</li>
        </ul>
      </div>
      <nav class="box">
        <ul>
          <li>
            <a href="#introduction">Introduction</a>
          </li>
          <li>
            <a href="#synopsis">Synopsis</a>
          </li>
          <li>
            <a href="#hierarchy">Hierarchy</a>
          </li>
          <li>
            <a href="#members">Members</a>
          </li>
          <li>
            <a href="#methods">Methods</a>
          </li>
          <li>
            <a href="../source/ar/www/html/vendor/symfony/options-resolver/Exception/NoSuchOptionException.php.xhtml#line24">Source</a>
          </li>
        </ul>
      </nav>
      <section>
        <h1 id="introduction"><small>Symfony\Component\OptionsResolver\Exception\</small>NoSuchOptionException</h1>
        <h4>Thrown when trying to read an option that has no value set.</h4>
        <p>When accessing optional options from within a lazy option or normalizer you should first<br/>check whether the optional option is set. You can do this with `isset($options['optional'])`.<br/>In contrast to the , this is a runtime exception that can<br/>occur when evaluating lazy options.<br/></p>
        <ul>
          <li>Author: Tobias Schultze &lt;http://tobion.de&gt;</li>
        </ul>
        <h2 id="synopsis">Synopsis</h2>
        <div class="synopsis">class NoSuchOptionException
                extends <a title="OutOfBoundsException" href="../classes/OutOfBoundsException.xhtml">OutOfBoundsException</a>
                implements
                <a title="Symfony\Component\OptionsResolver\Exception\ExceptionInterface" href="../interfaces/Symfony_Component_OptionsResolver_Exception_ExceptionInterface.xhtml">ExceptionInterface</a>
            {<br/><ul class="none"><li>// Inherited members from <span title="OutOfBoundsException">OutOfBoundsException</span></li><li>protected  <a href="#members">$message</a>;
                        </li><li>protected  <a href="#members">$code</a>;
                        </li><li>protected  <a href="#members">$file</a>;
                        </li><li>protected  <a href="#members">$line</a>;
                        </li></ul><ul class="none"><li>// Inherited members from <span title="RuntimeException">RuntimeException</span></li><li>protected  <a href="#members">$message</a>;
                        </li><li>protected  <a href="#members">$code</a>;
                        </li><li>protected  <a href="#members">$file</a>;
                        </li><li>protected  <a href="#members">$line</a>;
                        </li></ul><ul class="none"><li>// Inherited members from <span title="Exception">Exception</span></li><li>protected  <a href="#members">$message</a>;
                        </li><li>protected  <a href="#members">$code</a>;
                        </li><li>protected  <a href="#members">$file</a>;
                        </li><li>protected  <a href="#members">$line</a>;
                        </li></ul><ul class="none"><li>// Inherited methods from <span title="OutOfBoundsException">OutOfBoundsException</span></li><li>public final string <a title="OutOfBoundsException" href="../classes/OutOfBoundsException/getMessage.xhtml">getMessage</a>()
                        </li><li>public final Throwable <a title="OutOfBoundsException" href="../classes/OutOfBoundsException/getPrevious.xhtml">getPrevious</a>()
                        </li><li>public final mixed <a title="OutOfBoundsException" href="../classes/OutOfBoundsException/getCode.xhtml">getCode</a>()
                        </li><li>public final string <a title="OutOfBoundsException" href="../classes/OutOfBoundsException/getFile.xhtml">getFile</a>()
                        </li><li>public final int <a title="OutOfBoundsException" href="../classes/OutOfBoundsException/getLine.xhtml">getLine</a>()
                        </li><li>public final array <a title="OutOfBoundsException" href="../classes/OutOfBoundsException/getTrace.xhtml">getTrace</a>()
                        </li><li>public final string <a title="OutOfBoundsException" href="../classes/OutOfBoundsException/getTraceAsString.xhtml">getTraceAsString</a>()
                        </li><li>public string <a title="OutOfBoundsException" href="../classes/OutOfBoundsException/__toString.xhtml">__toString</a>()
                        </li></ul><ul class="none"><li>// Inherited methods from <span title="RuntimeException">RuntimeException</span></li><li>public final string <a title="RuntimeException" href="../classes/RuntimeException/getMessage.xhtml">getMessage</a>()
                        </li><li>public final Throwable <a title="RuntimeException" href="../classes/RuntimeException/getPrevious.xhtml">getPrevious</a>()
                        </li><li>public final mixed <a title="RuntimeException" href="../classes/RuntimeException/getCode.xhtml">getCode</a>()
                        </li><li>public final string <a title="RuntimeException" href="../classes/RuntimeException/getFile.xhtml">getFile</a>()
                        </li><li>public final int <a title="RuntimeException" href="../classes/RuntimeException/getLine.xhtml">getLine</a>()
                        </li><li>public final array <a title="RuntimeException" href="../classes/RuntimeException/getTrace.xhtml">getTrace</a>()
                        </li><li>public final string <a title="RuntimeException" href="../classes/RuntimeException/getTraceAsString.xhtml">getTraceAsString</a>()
                        </li><li>public string <a title="RuntimeException" href="../classes/RuntimeException/__toString.xhtml">__toString</a>()
                        </li></ul><ul class="none"><li>// Inherited methods from <span title="Exception">Exception</span></li><li>public void <a title="Exception" href="../classes/Exception/__construct.xhtml">__construct</a>()
                        </li><li>public final string <a title="Exception" href="../classes/Exception/getMessage.xhtml">getMessage</a>()
                        </li><li>public final Throwable <a title="Exception" href="../classes/Exception/getPrevious.xhtml">getPrevious</a>()
                        </li><li>public final mixed <a title="Exception" href="../classes/Exception/getCode.xhtml">getCode</a>()
                        </li><li>public final string <a title="Exception" href="../classes/Exception/getFile.xhtml">getFile</a>()
                        </li><li>public final int <a title="Exception" href="../classes/Exception/getLine.xhtml">getLine</a>()
                        </li><li>public final array <a title="Exception" href="../classes/Exception/getTrace.xhtml">getTrace</a>()
                        </li><li>public final string <a title="Exception" href="../classes/Exception/getTraceAsString.xhtml">getTraceAsString</a>()
                        </li><li>public string <a title="Exception" href="../classes/Exception/__toString.xhtml">__toString</a>()
                        </li></ul>

            }<br/></div>
        <h2 id="hierarchy">Hierarchy</h2>
        <div class="styled">
          <h4>Extends</h4>
          <ul>
            <li>
              <a title="OutOfBoundsException" href="../classes/OutOfBoundsException.xhtml">OutOfBoundsException</a>
            </li>
          </ul>
          <h4>Implements</h4>
          <ul>
            <li>
              <a title="Symfony\Component\OptionsResolver\Exception\ExceptionInterface" href="../interfaces/Symfony_Component_OptionsResolver_Exception_ExceptionInterface.xhtml">Symfony\Component\OptionsResolver\Exception\ExceptionInterface</a>
            </li>
          </ul>
        </div>
        <h2 id="members">Members</h2>
        <div class="styled members">
          <h4>protected</h4>
          <ul class="members">
            <li id="code"><strong>$code</strong>
                —
                int<br/><span class="indent">The exception code</span></li>
            <li id="code"><strong>$code</strong>
                —
                int<br/><span class="indent">The exception code</span></li>
            <li id="code"><strong>$code</strong>
                —
                int<br/><span class="indent">The exception code</span></li>
            <li id="file"><strong>$file</strong>
                —
                string<br/><span class="indent">The filename where the exception was created</span></li>
            <li id="file"><strong>$file</strong>
                —
                string<br/><span class="indent">The filename where the exception was created</span></li>
            <li id="file"><strong>$file</strong>
                —
                string<br/><span class="indent">The filename where the exception was created</span></li>
            <li id="line"><strong>$line</strong>
                —
                int<br/><span class="indent">The line where the exception was created</span></li>
            <li id="line"><strong>$line</strong>
                —
                int<br/><span class="indent">The line where the exception was created</span></li>
            <li id="line"><strong>$line</strong>
                —
                int<br/><span class="indent">The line where the exception was created</span></li>
            <li id="message"><strong>$message</strong>
                —
                string<br/><span class="indent">The exception message</span></li>
            <li id="message"><strong>$message</strong>
                —
                string<br/><span class="indent">The exception message</span></li>
            <li id="message"><strong>$message</strong>
                —
                string<br/><span class="indent">The exception message</span></li>
          </ul>
        </div>
        <h2 id="methods">Methods</h2>
        <div class="styled">
          <h3>Inherited from <a title="OutOfBoundsException" href="../classes/OutOfBoundsException.xhtml">OutOfBoundsException</a></h3>
          <h4>public</h4>
          <ul>
            <li id="__toString">
              <a title="OutOfBoundsException" href="../classes/OutOfBoundsException/__toString.xhtml">__toString()</a>
            </li>
            <li id="getCode">
              <a title="OutOfBoundsException" href="../classes/OutOfBoundsException/getCode.xhtml">getCode()</a>
            </li>
            <li id="getFile">
              <a title="OutOfBoundsException" href="../classes/OutOfBoundsException/getFile.xhtml">getFile()</a>
            </li>
            <li id="getLine">
              <a title="OutOfBoundsException" href="../classes/OutOfBoundsException/getLine.xhtml">getLine()</a>
            </li>
            <li id="getMessage">
              <a title="OutOfBoundsException" href="../classes/OutOfBoundsException/getMessage.xhtml">getMessage()</a>
            </li>
            <li id="getPrevious">
              <a title="OutOfBoundsException" href="../classes/OutOfBoundsException/getPrevious.xhtml">getPrevious()</a>
            </li>
            <li id="getTrace">
              <a title="OutOfBoundsException" href="../classes/OutOfBoundsException/getTrace.xhtml">getTrace()</a>
            </li>
            <li id="getTraceAsString">
              <a title="OutOfBoundsException" href="../classes/OutOfBoundsException/getTraceAsString.xhtml">getTraceAsString()</a>
            </li>
          </ul>
          <h3>Inherited from <a title="RuntimeException" href="../classes/RuntimeException.xhtml">RuntimeException</a></h3>
          <h4>public</h4>
          <ul>
            <li id="__toString">
              <a title="RuntimeException" href="../classes/RuntimeException/__toString.xhtml">__toString()</a>
            </li>
            <li id="getCode">
              <a title="RuntimeException" href="../classes/RuntimeException/getCode.xhtml">getCode()</a>
            </li>
            <li id="getFile">
              <a title="RuntimeException" href="../classes/RuntimeException/getFile.xhtml">getFile()</a>
            </li>
            <li id="getLine">
              <a title="RuntimeException" href="../classes/RuntimeException/getLine.xhtml">getLine()</a>
            </li>
            <li id="getMessage">
              <a title="RuntimeException" href="../classes/RuntimeException/getMessage.xhtml">getMessage()</a>
            </li>
            <li id="getPrevious">
              <a title="RuntimeException" href="../classes/RuntimeException/getPrevious.xhtml">getPrevious()</a>
            </li>
            <li id="getTrace">
              <a title="RuntimeException" href="../classes/RuntimeException/getTrace.xhtml">getTrace()</a>
            </li>
            <li id="getTraceAsString">
              <a title="RuntimeException" href="../classes/RuntimeException/getTraceAsString.xhtml">getTraceAsString()</a>
            </li>
          </ul>
          <h3>Inherited from <a title="Exception" href="../classes/Exception.xhtml">Exception</a></h3>
          <h4>public</h4>
          <ul>
            <li id="__toString"><a title="Exception" href="../classes/Exception/__toString.xhtml">__toString()</a>
                — String representation of the exception</li>
            <li id="getCode"><a title="Exception" href="../classes/Exception/getCode.xhtml">getCode()</a>
                — Gets the Exception code</li>
            <li id="getFile"><a title="Exception" href="../classes/Exception/getFile.xhtml">getFile()</a>
                — Gets the file in which the exception was created</li>
            <li id="getLine"><a title="Exception" href="../classes/Exception/getLine.xhtml">getLine()</a>
                — Gets the line in which the exception was created</li>
            <li id="getMessage"><a title="Exception" href="../classes/Exception/getMessage.xhtml">getMessage()</a>
                — Gets the Exception message</li>
            <li id="getPrevious"><a title="Exception" href="../classes/Exception/getPrevious.xhtml">getPrevious()</a>
                — Returns previous Exception</li>
            <li id="getTrace"><a title="Exception" href="../classes/Exception/getTrace.xhtml">getTrace()</a>
                — Gets the stack trace</li>
            <li id="getTraceAsString"><a title="Exception" href="../classes/Exception/getTraceAsString.xhtml">getTraceAsString()</a>
                — Gets the stack trace as a string</li>
          </ul>
        </div>
      </section>
    </div>
    <footer>
      <span>Generated using phpDox 0.12.0 - Copyright (C) 2010 - 2022 by Arne Blankerts and Contributors</span>
    </footer>
  </body>
</html>
